test/framework/src/main/java/org/elasticsearch/test/disruption/DisruptableMockTransport.java

```diff
@@ -94,7 +95,12 @@ public String toString() {
             public void run() {
                 switch (getConnectionStatus(getLocalNode(), destination)) {
                     case BLACK_HOLE:
-                        logger.trace("dropping {}", requestDescription);
+                        if (action.equals(HANDSHAKE_ACTION_NAME)) {
```

Comment:
This makes me unhappy, but I was unable to come up with a better idea. Thoughts?

Reply:
I've pushed a7a76c09219 which refactors the class a bit to make it more extensible. Let me know if you like that better.